### PR TITLE
More UBSAN fixes

### PR DESCRIPTION
**Pull request by MasterDuke17**

### CODE REVIEW

File: src/6model/serialization.c [hunk 1]
```diff
@@ -901,8 +901,9 @@ static MVMObject * concatenate_outputs(MVMThreadContext *tc, MVMSerializationWri
     /* Put repossessions table in place, and set location/rows in header. */
     write_int32(output, 56, offset);
     write_int32(output, 60, writer->root.num_repos);
-    memcpy(output + offset, writer->root.repos_table,
-        writer->root.num_repos * REPOS_TABLE_ENTRY_SIZE);
+    if (writer->root.repos_table != NULL)
```
Comment (niner) [inline src/6model/serialization.c:904]: There's no way we get here with repos_table being NULL, but I guess if it makes ubsan happy...
Could you add a comment that the check is just for ubsan? That way it won't trick readers into thinking that there is a case where there's no repos_table.

Comment (MasterDuke17) [reply src/6model/serialization.c:904]: This was a runtime warning from UBSAN, here's the stacktrace, it happened during a `make install` of NQP.
```
src/6model/serialization.c:904:9: runtime error: null pointer passed as argument 2, which is declared to never be null
    #0 0x731d31eea89c in concatenate_outputs src/6model/serialization.c:904
    #1 0x731d322bad42 in MVM_serialization_serialize src/6model/serialization.c:1426
    #2 0x731d31f8d5d2 in MVM_interp_run src/core/interp.c:5334
    #3 0x558ea3087c8b in main src/main.c:287
    #4 0x731d31634e07  (/usr/lib/libc.so.6+0x25e07) (BuildId: 98b3d8e0b8c534c769cb871c438b4f8f3a8e4bf3)
    #5 0x731d31634ecb in __libc_start_main (/usr/lib/libc.so.6+0x25ecb) (BuildId: 98b3d8e0b8c534c769cb871c438b4f8f3a8e4bf3)
    #6 0x558ea3087fc4 in _start (/home/dan/Source/perl6/install/bin/moar+0x2fc4) (BuildId: f682bce564648b38c51e2d28e1984ef849e6c654)
```

Comment (MasterDuke17) [reply src/6model/serialization.c:904]: And some higher-level backtraces:
```
[dan@alexandria nqp]$ gdb --args '/home/dan/Source/perl6/install/bin/moar' --libpath=src/vm/moar/stage0 src/vm/moar/stage0/nqp.moarvm --bootstrap --setting-path=gen/moar/stage1 --module-path=gen/moar/stage1 --no-regex-lib --target=mbc --setting=NQPCORE --stable-sc=stage1 --output=gen/moar/stage1/MASTOps.moarvm /home/dan/Source/perl6/install/share/nqp/lib/MAST/Ops.nqp
GNU gdb (GDB) 15.2
Copyright (C) 2024 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /home/dan/Source/perl6/install/bin/moar...
(gdb) b serialization.c:904
T�)� `
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (serialization.c:904) pending.
(gdb) r
Starting program: /home/dan/Source/perl6/install/bin/moar --libpath=src/vm/moar/stage0 src/vm/moar/stage0/nqp.moarvm --bootstrap --setting-path=gen/moar/stage1 --module-path=gen/moar/stage1 --no-regex-lib --target=mbc --setting=NQPCORE --stable-sc=stage1 --output=gen/moar/stage1/MASTOps.moarvm /home/dan/Source/perl6/install/share/nqp/lib/MAST/Ops.nqp
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Downloading 5.56 M separate debug info for /usr/lib/libzstd.so.1
[New Thread 0x7ffff59ff6c0 (LWP 170969)]                                                                                                                                                                                                          
3rdparty/dynasm/dasm_x86.h:108:48: runtime error: applying non-zero offset 18446744073642442752 to null pointer
3rdparty/dynasm/dasm_x86.h:601:6: runtime error: store to misaligned address 0x7ffff7f81039 for type 'unsigned int', which requires 4 byte alignment
0x7ffff7f81039: note: pointer points here
 e2 49 8b  8e 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00
              ^ 
3rdparty/dynasm/dasm_x86.h:609:5: runtime error: store to misaligned address 0x7ffff7f800bb for type 'short unsigned int', which requires 2 byte alignment
0x7ffff7f800bb: note: pointer points here
 66  c7 40 0e 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00
              ^ 
src/disp/program.c:3292:17: runtime error: load of misaligned address 0x7ffff4eccb13 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4eccb13: note: pointer points here
 ff  7f 00 00 0c 00 0e 00 0b  00 30 fa bc f6 ff 7f 00  00 50 16 90 f6 ff 7f 00  00 30 9f 83 f6 ff 7f
              ^ 
src/disp/program.c:3330:17: runtime error: load of misaligned address 0x7ffff4eccb15 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4eccb15: note: pointer points here
 00 00 0c 00 0e 00 0b  00 30 fa bc f6 ff 7f 00  00 50 16 90 f6 ff 7f 00  00 30 9f 83 f6 ff 7f 00  00
             ^ 
src/disp/program.c:3432:69: runtime error: load of misaligned address 0x7ffff4eccb17 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4eccb17: note: pointer points here
 0c 00 0e 00 0b  00 30 fa bc f6 ff 7f 00  00 50 16 90 f6 ff 7f 00  00 30 9f 83 f6 ff 7f 00  00 50 16
             ^ 
src/core/args.c:501:5: runtime error: load of misaligned address 0x7ffff4ebe14f for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4ebe14f: note: pointer points here
 ff 7f 00 00 03  00 e0 a9 82 f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 50 16 90 f6 ff 7f 00  00 50 16
             ^ 
src/spesh/log.c:130:67: runtime error: load of misaligned address 0x7ffff4ebe189 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4ebe189: note: pointer points here
 ff 7f 00  00 03 00 e0 a9 82 f6 ff  7f 00 00 30 fa bc f6 ff  7f 00 00 50 16 90 f6 ff  7f 00 00 50 16
              ^ 
src/spesh/arg_guard.c:484:44: runtime error: load of misaligned address 0x7ffff4ebe14f for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4ebe14f: note: pointer points here
 ff 7f 00 00 03  00 e0 a9 82 f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 50 16 90 f6 ff 7f 00  00 50 16
             ^ 
src/disp/syscall.c:717:22: runtime error: load of misaligned address 0x7ffff4eac2c3 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4eac2c3: note: pointer points here
 ff  7f 00 00 02 00 50 16 90  f6 ff 7f 00 00 05 00 06  00 03 00 02 00 00 a1 70  f6 ff 7f 00 00 70 42
              ^ 
src/disp/program.c:3305:17: runtime error: load of misaligned address 0x7ffff4e8f81d for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4e8f81d: note: pointer points here
 ff 7f 00 00 13 00 b0  54 90 f6 ff 7f 00 00 50  16 90 f6 ff 7f 00 00 02  00 e0 a9 82 f6 ff 7f 00  00
             ^ 
src/6model/reprs/MVMCapture.c:128:51: runtime error: load of misaligned address 0x7ffff4ebe607 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4ebe607: note: pointer points here
 ff 7f 00 00 3c  00 13 00 3a 00 08 00 06  00 50 16 90 f6 ff 7f 00  00 50 63 98 f6 ff 7f 00  00 a0 b5
             ^ 
src/core/args.c:513:5: runtime error: load of misaligned address 0x7ffff4ebe635 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4ebe635: note: pointer points here
 3a 00 38 00 2a 00 e0  a9 82 f6 ff 7f 00 00 30  fa bc f6 ff 7f 00 00 b0  c2 d7 f6 ff 7f 00 00 30  fa
             ^ 
src/core/args.c:507:5: runtime error: load of misaligned address 0x7ffff4e7d247 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4e7d247: note: pointer points here
 5a 00 59 00 58  00 e0 a9 82 f6 ff 7f 00  00 e0 53 98 f6 ff 7f 00  00 b0 97 aa f6 ff 7f 00  00 00 a1
             ^ 
src/disp/program.c:3562:77: runtime error: load of misaligned address 0x7ffff4ebe60d for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4ebe60d: note: pointer points here
 13 00 3a 00 08 00 06  00 50 16 90 f6 ff 7f 00  00 50 63 98 f6 ff 7f 00  00 a0 b5 98 f6 ff 7f 00  00
             ^ 
src/disp/syscall.c:702:22: runtime error: load of misaligned address 0x7ffff4e5843d for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4e5843d: note: pointer points here
 ff 7f 00 00 21 00 50  16 90 f6 ff 7f 00 00 90  6a 98 f6 ff 7f 00 00 30  fa bc f6 ff 7f 00 00 30  34
             ^ 
src/core/interp.c:5955:62: runtime error: load of misaligned address 0x7ffff4e5884f for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4e5884f: note: pointer points here
 ff 7f 00 00 21  00 3a 00 25 00 e0 a9 82  f6 ff 7f 00 00 30 fa bc  f6 ff 7f 00 00 40 61 9a  f6 ff 7f
             ^ 
src/disp/program.c:3298:17: runtime error: load of misaligned address 0x7ffff4eab365 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4eab365: note: pointer points here
 ff 7f 00 00 08 00 b0  54 90 f6 ff 7f 00 00 e0  51 9c f6 ff 7f 00 00 50  16 90 f6 ff 7f 00 00 50  16
             ^ 
src/core/args.c:95:45: runtime error: load of misaligned address 0x7ffff4e0f4db for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4e0f4db: note: pointer points here
 07  00 05 00 03 00 02 00 01  00 30 fa bc f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 b0 c2 d7 f6 ff 7f
              ^ 
src/core/args.c:241:62: runtime error: load of misaligned address 0x7ffff4e0f4d5 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4e0f4d5: note: pointer points here
 ff 7f 00 00 06 00 07  00 05 00 03 00 02 00 01  00 30 fa bc f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00
             ^ 
src/core/args.c:189:45: runtime error: load of misaligned address 0x7ffff4e0f4db for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4e0f4db: note: pointer points here
 07  00 05 00 03 00 02 00 01  00 30 fa bc f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 b0 c2 d7 f6 ff 7f
              ^ 
src/core/args.c:225:66: runtime error: load of misaligned address 0x7ffff4e0f4dd for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4e0f4dd: note: pointer points here
 05 00 03 00 02 00 01  00 30 fa bc f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 b0 c2 d7 f6 ff 7f 00  00
             ^ 
src/core/args.c:543:5: runtime error: load of misaligned address 0x7ffff4e05a51 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4e05a51: note: pointer points here
 07 00 04  00 11 00 e0 a9 82 f6 ff  7f 00 00 50 16 90 f6 ff  7f 00 00 e0 52 92 f6 ff  7f 00 00 00 a1
              ^ 
src/core/args.c:81:45: runtime error: load of misaligned address 0x7ffff4df1d55 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4df1d55: note: pointer points here
 01 00 05 00 04 00 30  fa bc f6 ff 7f 00 00 b0  54 90 f6 ff 7f 00 00 30  fa bc f6 ff 7f 00 00 0e  00
             ^ 
src/core/args.c:146:45: runtime error: load of misaligned address 0x7ffff4df1d55 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4df1d55: note: pointer points here
 01 00 05 00 04 00 30  fa bc f6 ff 7f 00 00 b0  54 90 f6 ff 7f 00 00 30  fa bc f6 ff 7f 00 00 0e  00
             ^ 
src/disp/syscall.c:628:22: runtime error: load of misaligned address 0x7ffff4df1d6f for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4df1d6f: note: pointer points here
 ff 7f 00 00 0e  00 50 16 90 f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 06 00
             ^ 
src/disp/syscall.c:660:22: runtime error: load of misaligned address 0x7ffff4da3907 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4da3907: note: pointer points here
 ff 7f 00 00 09  00 50 16 90 f6 ff 7f 00  00 30 9f 83 f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 30 9f
             ^ 
src/core/interp.c:5979:62: runtime error: load of misaligned address 0x7ffff4d2d1f7 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4d2d1f7: note: pointer points here
 11 00 02 00 0f  00 18 00 e0 a9 82 f6 ff  7f 00 00 30 fa bc f6 ff  7f 00 00 23 00 02 00 08  00 e0 a9
             ^ 
src/core/interp.c:5961:64: runtime error: load of misaligned address 0x7ffff4d2d1f9 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4d2d1f9: note: pointer points here
 02 00 0f  00 18 00 e0 a9 82 f6 ff  7f 00 00 30 fa bc f6 ff  7f 00 00 23 00 02 00 08  00 e0 a9 82 f6
              ^ 
src/core/args.c:608:5: runtime error: load of misaligned address 0x7ffff4d2d20f for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4d2d20f: note: pointer points here
 23 00 02 00 08  00 e0 a9 82 f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 50 16
             ^ 
src/core/args.c:1067:5: runtime error: load of misaligned address 0x7ffff4d2d20d for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4d2d20d: note: pointer points here
 00 00 23 00 02 00 08  00 e0 a9 82 f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00 30 fa bc f6 ff 7f 00  00
             ^ 
src/6model/reprs/P6int.c:63:52: runtime error: member access within misaligned address 0x052e278e5484 for type 'struct MVMP6intBody', which requires 8 byte alignment
0x052e278e5484: note: pointer points here
  00 00 00 00 00 00 00 00  f0 60 8e 27 2e 05 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00
              ^ 
src/disp/syscall.c:928:22: runtime error: load of misaligned address 0x7ffff4cdf22f for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4cdf22f: note: pointer points here
 ff 7f 00 00 03  00 20 23 9b f6 ff 7f 00  00 c0 b0 70 f6 ff 7f 00  00 70 42 83 f6 ff 7f 00  00 00 00
             ^ 
src/6model/reprs/P6int.c:83:52: runtime error: member access within misaligned address 0x052e27a40e74 for type 'struct MVMP6intBody', which requires 8 byte alignment
0x052e27a40e74: note: pointer points here
  30 00 00 00 00 00 00 00  58 0c a4 27 2e 05 00 00  78 0c a4 27 2e 05 00 00  b0 0c a4 27 2e 05 00 00
              ^ 
src/6model/reprs/P6int.c:93:54: runtime error: member access within misaligned address 0x052e260634bc for type 'struct MVMP6intBody', which requires 8 byte alignment
0x052e260634bc: note: pointer points here
  3c 00 00 00 3c 00 00 00  00 34 06 26 2e 05 00 00  20 34 06 26 2e 05 00 00  58 34 06 26 2e 05 00 00
              ^ 
src/core/args.c:590:5: runtime error: load of misaligned address 0x7ffff4ca2181 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4ca2181: note: pointer points here
 03 00 01  00 02 00 e0 a9 82 f6 ff  7f 00 00 00 a1 70 f6 ff  7f 00 00 70 42 83 f6 ff  7f 00 00 00 00
              ^ 
src/core/args.c:549:5: runtime error: load of misaligned address 0x7ffff4c8aa93 for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4c8aa93: note: pointer points here
 00  00 0b 00 09 00 e0 a9 82  f6 ff 7f 00 00 30 fa bc  f6 ff 7f 00 00 08 00 02  00 e0 a9 82 f6 ff 7f
              ^ 
src/core/args.c:537:5: runtime error: load of misaligned address 0x7ffff4c10acf for type 'MVMuint16', which requires 2 byte alignment
0x7ffff4c10acf: note: pointer points here
 00 00 00 00 17  00 11 00 0e 00 1b 00 e0  a9 82 f6 ff 7f 00 00 00  a1 70 f6 ff 7f 00 00 70  42 83 f6
             ^ 
src/6model/reprs/P6int.c:73:54: runtime error: member access within misaligned address 0x052e266b40b4 for type 'struct MVMP6intBody', which requires 8 byte alignment
0x052e266b40b4: note: pointer points here
  00 00 00 00 01 00 00 00  e8 d1 4c 31 2e 05 00 00  00 00 00 00 00 00 00 00  40 ec 6f 31 2e 05 00 00
              ^ 

Thread 1 "moar" hit Breakpoint 1, concatenate_outputs (tc=tc@entry=0x52e26020080, writer=writer@entry=0x52e2666ec80, type=type@entry=0x52e26601ee8) at src/6model/serialization.c:904
904         if (writer->root.repos_table == NULL)
(gdb) p writer->root.repos_table
$1 = 0x0
(gdb) next
905             fprintf(stderr, "uh oh, writer->root.repos_table is NULL!\n");
(gdb) call MVM_dump_backtrace(tc)
   at NQP::src/vm/moar/QAST/QASTCompilerMAST.nqp:878  (src/vm/moar/stage0/QAST.moarvm:serialize_sc)
 from NQP::src/vm/moar/QAST/QASTCompilerMAST.nqp:883  (src/vm/moar/stage0/QAST.moarvm:deserialization_code)
 from NQP::src/vm/moar/QAST/QASTCompilerMAST.nqp:824  (src/vm/moar/stage0/QAST.moarvm:compile_node)
 from NQP::src/vm/moar/QAST/QASTCompilerMAST.nqp:658  (src/vm/moar/stage0/QAST.moarvm:as_mast)
 from NQP::src/vm/moar/QAST/QASTCompilerMAST.nqp:345  (src/vm/moar/stage0/QAST.moarvm:to_mast)
 from NQP::src/vm/moar/QAST/QASTCompilerMAST.nqp:2175  (src/vm/moar/stage0/QAST.moarvm:to_mast)
 from NQP::src/vm/moar/HLL/Backend.nqp:790  (src/vm/moar/stage0/NQPHLL.moarvm:mast)
 from NQP::src/HLL/Compiler.nqp:462  (src/vm/moar/stage0/NQPHLL.moarvm:execute_stage)
 from NQP::src/HLL/Compiler.nqp:501  (src/vm/moar/stage0/NQPHLL.moarvm:run)
 from NQP::src/HLL/Compiler.nqp:504  (src/vm/moar/stage0/NQPHLL.moarvm:)
 from NQP::src/HLL/Compiler.nqp:496  (src/vm/moar/stage0/NQPHLL.moarvm:compile)
 from NQP::src/HLL/Compiler.nqp:166  (src/vm/moar/stage0/NQPHLL.moarvm:eval)
 from NQP::src/HLL/Compiler.nqp:441  (src/vm/moar/stage0/NQPHLL.moarvm:evalfiles)
 from NQP::src/HLL/Compiler.nqp:331  (src/vm/moar/stage0/NQPHLL.moarvm:command_eval)
 from NQP::src/HLL/Compiler.nqp:289  (src/vm/moar/stage0/NQPHLL.moarvm:command_line)
 from NQP::src/NQP/Compiler.nqp:37  (src/vm/moar/stage0/nqp.moarvm:MAIN)
 from NQP::src/NQP/Compiler.nqp:35  (src/vm/moar/stage0/nqp.moarvm:<mainline>)
 from <unknown>:1  (src/vm/moar/stage0/nqp.moarvm:<main>)
 from <unknown>:1  (src/vm/moar/stage0/nqp.moarvm:<entry>)
(gdb) call MVM_dump_bytecode(tc)
src/core/bytecodedump.c:257:29: runtime error: load of misaligned address 0x7ffff4f7c656 for type 'MVMuint32', which requires 4 byte alignment
0x7ffff4f7c656: note: pointer points here
 04 00 05 00 62 07  00 00 01 00 87 01 05 00  06 00 07 00 07 00 87 07  00 00 3e 03 06 00 2b 04  00 00
             ^ 
src/core/bytecodedump.c:258:38: runtime error: load of misaligned address 0x7ffff4f7c656 for type 'MVMuint32', which requires 4 byte alignment
0x7ffff4f7c656: note: pointer points here
 04 00 05 00 62 07  00 00 01 00 87 01 05 00  06 00 07 00 07 00 87 07  00 00 3e 03 06 00 2b 04  00 00
             ^ 
00000      checkarity         2, 2
00001      param_rp_o         loc_0_obj, 0
00002      param_rp_o         loc_1_obj, 1
00003      paramnamesused     
     annotation: NQP::src/vm/moar/QAST/QASTCompilerMAST.nqp:877
00004      decont             loc_4_obj, loc_0_obj
00005      wval               loc_5_obj, 0, 90
00006      getattr_o          loc_6_obj, loc_4_obj, loc_5_obj, '$!mast_compunit', 1
00007      decont             loc_5_obj, loc_6_obj
00008      const_s            loc_7_str, 'writer'
00009      dispatch_o         loc_6_obj, 'lang-meth-call', Callsite_0, loc_5_obj, loc_7_str, loc_6_obj
00010      decont             loc_5_obj, loc_6_obj
00011      const_s            loc_7_str, 'string-heap'
00012      dispatch_o         loc_6_obj, 'lang-meth-call', Callsite_0, loc_5_obj, loc_7_str, loc_6_obj
00013      decont             loc_5_obj, loc_6_obj
00014      const_s            loc_7_str, 'orig-strings'
00015      dispatch_o         loc_6_obj, 'lang-meth-call', Callsite_0, loc_5_obj, loc_7_str, loc_6_obj
00016   -> set                loc_2_obj, loc_6_obj
     annotation: NQP::src/vm/moar/QAST/QASTCompilerMAST.nqp:878
00017      wval               loc_6_obj, 4, 3
00018      serializetobuf     loc_6_obj, loc_1_obj, loc_2_obj, loc_6_obj
00019      set                loc_3_obj, loc_6_obj
     annotation: NQP::src/vm/moar/QAST/QASTCompilerMAST.nqp:879
00020      hlllist            loc_6_obj
00021      create             loc_6_obj, loc_6_obj
00022      const_i64_16       loc_8_int, 3
00023      setelemspos        loc_6_obj, loc_8_int
00024      const_i64_16       loc_8_int, 0
00025      setelemspos        loc_6_obj, loc_8_int
00026      push_o             loc_6_obj, loc_3_obj
00027      push_o             loc_6_obj, loc_2_obj
00028      const_i64_16       loc_8_int, 0
00029      hllboxtype_i       loc_5_obj
00030      box_i              loc_5_obj, loc_8_int, loc_5_obj
00031      push_o             loc_6_obj, loc_5_obj
00032      return_o           loc_6_obj
(gdb)
```

Comment (niner) [reply src/6model/serialization.c:904]: Aaah...I missed the early return at https://github.com/MoarVM/MoarVM/blob/main/src/6model/serialization.c#L1307
If that triggers, repos_table is indeed NULL.

But shouldn't we then also guard the MVM_free here?
https://github.com/MoarVM/MoarVM/blob/main/src/6model/serialization.c#L1440

Comment (MasterDuke17) [reply src/6model/serialization.c:904]: I don't think it's necessary, `free(NULL)` is allowed.

Comment (niner) [reply src/6model/serialization.c:904]: Ah, right.

**Review by niner — APPROVED** [review]

**Pull request merged**